syntax = "proto3";

message Answer{
  string date_send = 1;
  int64 id = 2;
  int64 id_team = 3;
  int64 id_user = 4;
  int64 id_task = 5;
  int64 id_contest = 6;
  string name_compilation = 8;
  string total = 9;
  string time = 10;
  string memory_size = 11;
  int32 number_test = 12;
  int32 points = 13;
}

message SendAnswerRequest{
  int64 id_task = 1;
  int64 id_user = 2;
  int64 id_team = 3;
  int64 id_contest = 4;
  int64 id_compiler = 5;
  bytes program_file = 6;
}

message SendAnswerCodeResponse{
  string code = 1;
}

message GetListAnswersTaskRequest{
  string type_contest = 1;
  int64 id = 2;
  int64 id_task = 3;
}

message GetListAnswersTaskResponse{
  repeated Answer answers = 1;
}

message GetAnswersContestRequest{
  int64 id_contest = 1;
}

message GetAnswersContestResponse{
  repeated Answer answers = 1;
}

message GetReportFileRequest{
  int64 id_answer = 1;
}

message GetReportFileResponse{
  bytes report_file = 1;
}

service AnswerApi{
  rpc SendAnswer(SendAnswerRequest) returns (SendAnswerCodeResponse){}
  rpc GetListAnswersTask(GetListAnswersTaskRequest) returns (GetListAnswersTaskResponse){}
  rpc GetAnswersContest(GetAnswersContestRequest) returns (GetAnswersContestResponse){}
  rpc GetReportFile(GetReportFileRequest) returns (GetReportFileResponse){}
}